### PR TITLE
PyGRB writing/loading timeslides

### PR DESCRIPTION
**Pull request by jakeb245**
This PR is the first part of updating `pycbc_pygrb_efficiency` (see #4419), although that file is not directly changed here.

The `load_time_slides` function in `pygrb_postprocessing_utils` was dependent on XML output files. This PR updates that function to use HDF files. To make this happen, I had to update `pycbc_multi_inspiral` and the `EventManagerCoherent` class to write the full timeslide information to the output file. This information is stored in the `network` group. It can be accessed through the path `network` -> `timeslides` -> IFOs -> slide shifts. I am open to restructuring this if requested.

The `load_time_slides` function should return the timeslide information in the same format as it was written for XML tables. 

I also found `pycbc_multi_inspiral` was trying to write the `slide_id` in both the network and IFO groups. It was failing to write the slide ids to the IFO groups because those outputs are hardcoded in the`write_to_hdf` function (but the network outputs are not). I decided to change the IFO groups to have the slide shift written instead of the slide ids again.

This might be useful for updating the other scripts as described in the issue linked above, so I think it warrants its own PR.

### CODE REVIEW

File: bin/pycbc_multi_inspiral [hunk 1]
```diff
@@ -268,7 +268,7 @@ with ctx:
         'bank_chisq': float32,
         'bank_chisq_dof': int,
         'cont_chisq': float32,
-        'slide_id': int
```
Comment (spxiwh) [inline bin/pycbc_multi_inspiral:271]: I would be very cautious in making this change! You will likely want to do things like `slide_id == specific_slide_id` in what follows, and `slide_shift == specific_slide_shift` is a "bad idea TM" (https://www.geeksforgeeks.org/problem-in-comparing-floating-point-numbers-and-how-to-compare-them-correctly/) ... While you can implement "approximate equality" it is slower and can be more ugly (although I think numpy has "almost equal" functions built in for this).

Comment (pannarale) [reply bin/pycbc_multi_inspiral:271]: @jakeb245 was there a specific problem you were trying to solve with this change?

Comment (jakeb245) [reply bin/pycbc_multi_inspiral:271]: Previously there was no information about the slide shifts being written to the output files (the `slide_id` that was previously there was not written due to some hardcoding). I thought putting the slide shift associated with each trigger would be a good way to fix this. I eventually went with writing the `timeslides` column instead.

I think I will revert this back to `slide_id`, and make sure it gets written to the output file properly.

**Comment by jakeb245** [comment]
I've finally gotten to run these changes in the new workflow, and it seems that `pycbc_grb_trig_combiner` and likely `pycbc_grb_trig_cluster` will need updates to process time slide information, so I'm moving this PR to draft status until I have something for that.

**Comment by pannarale** [comment]
We will hold up on merging this until it is tested in a complete workflow with all updates relevant to https://github.com/gwastro/pycbc/issues/4419

**Review by pannarale — APPROVED** [review]

**Pull request merged**